### PR TITLE
WIP: add MDNS Repeater

### PR DESCRIPTION
**Pull request by fabianfrz**
everything except the service template works.
It seems to be not recognised by the service script.

### CODE REVIEW

File: net/mdns-repeater/src/etc/inc/plugins.inc.d/mdnsrepeater.inc [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    return false;
+}
+
+function mdnsrepeater_firewall($fw)
+{
+    if (!mdnsrepeater_enabled()) {
+        return;
+    }
```
Comment (fabianfrz) [inline net/mdns-repeater/src/etc/inc/plugins.inc.d/mdnsrepeater.inc]: if required: open udp/5353

**Comment by fichtner** [comment]
<3

**Comment by fabianfrz** [comment]
@fichtner do you have an idea, why the service configuration file is not loaded?

File: net/mdns-repeater/Makefile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+PLUGIN_NAME=       mdns-repeater
+PLUGIN_VERSION=    1.0
+PLUGIN_COMMENT=    Start, stop, configure mdns-repater
```
Comment (fichtner) [inline net/mdns-repeater/Makefile]: this could explain better what it does, like "Allow multicast DNS between networks" maybe?

Comment (fabianfrz) [reply net/mdns-repeater/Makefile]: good idea

File: net/mdns-repeater/src/opnsense/service/templates/OPNsense/mdnsrepeater/+TARGETS [hunk 3]
```diff
@@ -0,0 +1 @@
+mdnsrepeater:/etc/rc.conf.d/mdnsrepeater
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/service/templates/OPNsense/mdnsrepeater/+TARGETS]: Can you keep the namespace the same? OPNsense/MDNSRepeater

File: net/mdns-repeater/src/opnsense/service/templates/OPNsense/mdnsrepeater/mdnsrepeater [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{% if helpers.exists('OPNsense.mdnsrepeater.enabled') and OPNsense.mdnsrepeater.enabled == '1' %}
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/service/templates/OPNsense/mdnsrepeater/mdnsrepeater]: Preferably also here... OPNsense.MDNSRepeater

File: net/mdns-repeater/src/opnsense/service/templates/OPNsense/mdnsrepeater/mdnsrepeater [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+{% from 'OPNsense/Macros/interface.macro' import physical_interface %}
+mdns_repeater_enable="YES"
+{% set osifnames = OPNsense.mdnsrepeater.interfaces.split(',') %}
+mdns_repeater_interfaces="{% for i in osifnames %}{{ physical_interface(i) }} {% endfor %}"
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/service/templates/OPNsense/mdnsrepeater/mdnsrepeater]: join() is a bit more elegant, see:

https://github.com/opnsense/core/blob/master/src/opnsense/service/templates/OPNsense/IDS/rc.conf.d#L16

**Comment by fichtner** [comment]
does the template work now?

**Comment by fabianfrz** [comment]
spectacle ns8560
same error

File: net/mdns-repeater/src/opnsense/service/templates/OPNsense/MDNSRepeater/+TARGETS [hunk 6]
```diff
@@ -0,0 +1 @@
+mdnsrepeater:/etc/rc.conf.d/mdnsrepeater
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/service/templates/OPNsense/MDNSRepeater/+TARGETS]: try "mdns_repeater"

Comment (fabianfrz) [reply net/mdns-repeater/src/opnsense/service/templates/OPNsense/MDNSRepeater/+TARGETS]: looks good now

File: net/mdns-repeater/src/etc/inc/plugins.inc.d/mdnsrepeater.inc [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    $services = array();
+
+    // don't load the service if it is not enabled
+    // maybe there are no settings yet
+    if (!mdnsrepeater_enabled()) {
+        return $services;
+    }
+
+    $services[] = array(
+        'description' => gettext('MDNS Repeater (MDNS Proxy)'),
```
Comment (fichtner) [inline net/mdns-repeater/src/etc/inc/plugins.inc.d/mdnsrepeater.inc]: one of those is enough, better to use the underlying service name

File: net/mdns-repeater/src/opnsense/mvc/app/controllers/OPNsense/MDNSRepeater/Api/ServiceController.php [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+use \OPNsense\Core\Backend;
+use \OPNsense\MDNSRepeater\MDNSRepeater;
+
+class ServiceController extends ApiControllerBase
+{
+    public function statusAction()
+    {
+        $backend = new Backend();
+        $result = array('result' => 'failed');
+        $mdlMDNSRepeater = new MDNSRepeater();
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/mvc/app/controllers/OPNsense/MDNSRepeater/Api/ServiceController.php]: unused

File: net/mdns-repeater/src/opnsense/mvc/app/controllers/OPNsense/MDNSRepeater/Api/ServiceController.php [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        $backend->configdRun('template reload OPNsense/MDNSRepeater');
+        $result['result'] = $backend->configdRun('mdnsrepeater start');
+        return $result;
+    }
+
+    public function stopAction()
+    {
+        $backend = new Backend();
+        $result = array("result" => "failed");
+        $mdlMDNSRepeater = new MDNSRepeater();
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/mvc/app/controllers/OPNsense/MDNSRepeater/Api/ServiceController.php]: unused

File: net/mdns-repeater/src/opnsense/mvc/app/controllers/OPNsense/MDNSRepeater/Api/SettingsController.php [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+namespace OPNsense\MDNSRepeater\Api;
+
+use \OPNsense\Base\ApiMutableModelControllerBase;
+
+class SettingsController extends ApiMutableModelControllerBase
+{
+  static protected $internalModelClass = '\OPNsense\MDNSRepeater\MDNSRepeater';
+  static protected $internalModelName = 'mdnsrepeater';
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/mvc/app/controllers/OPNsense/MDNSRepeater/Api/SettingsController.php]: MDNSRepeater ?

**Comment by fabianfrz** [comment]
just keep it on master - so it can be tested. Also I have to update the service indicator to work - however this version seems to run.

**Comment by fichtner** [comment]
then please change it to PLUGIN_DEVEL=yes, I need to push it to stable/17.7 in any way to prevent the build from failing.

**Comment by joaquinvacas** [comment]
Is there any way of fine-tuning mDNS?

I mean, currently I am using Avahi reflector under OpenWRT, so I can filter which services should be "repeated" like this:
imagen



A good case use is to "repeat" printer's mDNS for auto-discover but not letting other services be repeated like TVs, Cast/AirPlay and so on.

**Pull request merged**